compiler/runtime/obstack.c

```diff
@@ -59,63 +59,63 @@ __thread char* end = NULL;
 
 typedef struct {
   char* head;
-  char* page;
+  struct sk_obstack* page;
   char* end;
 } sk_saved_obstack_t;
 
+typedef struct sk_obstack {
+  struct sk_obstack* previous;
+  size_t size;
+  sk_saved_obstack_t saved;
+  char contents[0];
```

Comment:
nit: the comments at the top of obstack.c use the term "user data" for this, maybe change either here or there to use the same term in both places.